### ISSUE TITLE
Simple 'hello world' app exits sbt session after `run` command

### ISSUE THREAD

**Issue by AugustNagro** [zio/zio#6895]
Hello, I have a simple hello-world app:

```scala
import zio.*

object Main extends ZIOAppDefault:
  def run = Console.printLine("hello world")
```

With these build settings:

```scala
val ZioVersion = "2.0.0-RC6"

inThisBuild(
  Seq(
    organization := "com.augustnagro",
    version := "0.1.0-SNAPSHOT",
    versionScheme := Some("early-semver"),
    scalaVersion := "3.1.2"
  )
)

lazy val root = project
  .in(file("."))
  .settings(
    name := "zio-exits-sbt-session-bug",
    libraryDependencies ++= Seq(
      "dev.zio" %% "zio" % ZioVersion,
      "org.scalameta" %% "munit" % "0.7.29" % Test
    )
  )
```

To develop it, I start sbt with the command `sbt`. Then, in the sbt session I type `run` to execute the program.

It does correctly print 'hello world', but it also shuts down the JVM sbt is using.

```
[info] running Main 
hello world
[info] shutting down sbt server
zio-exits-sbt-session-bug $ 
```

This issue doesn't depend on the Java version, I tried with Java 8 and 11.

Here is a reproducer: 

https://github.com/AugustNagro/zio-exits-sbt-session-bug

**Comment by adamgfraser** [comment]
@AugustNagro You need to use `fork in Run := true`.

**Comment by AugustNagro** [comment]
Thanks, `fork in Run := true` doesn't compile but `fork := true` works.

Sbt changed recently to use `/` instead of `in` (https://www.scala-sbt.org/1.x/docs/Forking.html). I think we'd want to fork in all cases anyway.

**Comment by AugustNagro** [comment]
Going to re-open, since `fork := true` doesn't work when needing to read from stdin.

```
import zio.*

object Main extends ZIOAppDefault:

  def run =
    for
      _ <- Console.printLine("pick a number 1-10")
      n <- Console.readLine
      _ <- Console.printLine(s"picked $n")
    yield ()

```
```
[info] running (fork) Main 
[info] pick a number 1-10
[info] timestamp=2022-06-02T15:26:00.317453Z level=ERROR thread=#zio-fiber-0 message="" cause="Exception in thread "zio-fiber-6" java.io.EOFException: java.io.EOFException: There is no more input left to read
[info] 	at <empty>.Main$.run.macro(Main.scala:8)
[info] 	at <empty>.Main$.run.macro(Main.scala:10)"
[error] Nonzero exit code returned from runner: 1
[error] (Compile / run) Nonzero exit code returned from runner: 1
[error] Total time: 7 s, completed Jun 2, 2022, 8:26:00 AM
```

**Comment by adamgfraser** [comment]
This really needs to be addressed at the SBT level since this was a regression introduced there but possibly we can figure out something to work around it.

**Comment by adamgfraser** [comment]
Closed by #6995.
